### DISCUSSION TITLE
BigQuery foreign table with JSON column : how to read/write

### DISCUSSION THREAD

**Discussion by lordlothar99** [supabase/supabase#30785] [Questions]
Hi !
I feel dumb to ask :
I have a BigQuery table with a JSON column.
How should it be mapped in Supabase when I create the foreign table ?
I tried types "text", "json", "jsonb"
But when I execute a SELECT from Supabase, I get "ERROR:  HV000: field type Json not supported"

Thanks !

**Comment by GaryAustin1** [comment]
https://supabase.com/docs/guides/database/extensions/wrappers/bigquery#supported-data-types
Not sure you can.

**Reply by lordlothar99** [reply]
yeah, that's what I saw as well ...
sounds super strange, but I guess the only option is to not use json type in BigQuery then ?

**Reply by GaryAustin1** [reply]
Unless someone from Supabase responds, that is how I read it.  Your tests sort of confirm that limitation.

**Reply by lordlothar99** [reply]
seems like I'm not the first one to face this issue : https://github.com/supabase/wrappers/issues/70
let's hope json support will come soon

**Reply by GaryAustin1** [reply]
Well that was 1.8 years ago...

**Reply by lordlothar99** [reply]
yep ...
the fastest is to change the type in BQ 😄
